app/main.py

```diff
@@ -1,6 +1,31 @@
+import os
+import requests
+from dotenv import load_dotenv
+
+load_dotenv()
+
+KEY = os.environ["API_KEY"]
+BASE_URL = "http://api.weatherapi.com/v1/current.json"
+CITY = "Paris"
+
+
 def get_weather() -> None:
-    # write your code here
-    pass
+    response = requests.get(
+        f"{BASE_URL}?key={KEY}&q={CITY}",
+    )
+
+    data = response.json()
+    city = data["location"]["name"]
+    country = data["location"]["country"]
+    current_data = data["location"]["localtime"]
+    temperature = data["current"]["temp_c"]
+    weather = data["current"]["condition"]["text"]
+    print(
+        f"{city}/{country} "
+        f"{current_data} "
+        f"Weather: {temperature} Celsius, "
+        f"{weather}"
+    )
```

Comment:
Why Use a try Block Here?
Network Errors: The requests.get call can fail due to network issues, timeouts, or invalid URLs. Wrapping it in a try block ensures that these errors are caught and handled gracefully rather than causing your program to crash.

HTTP Errors: If the API returns an error response (e.g., 4xx or 5xx status codes), using response.raise_for_status() will raise an exception. This exception is caught by the except block to provide a meaningful error message.

JSON Parsing Errors: The response.json() call can fail if the response is not valid JSON. Wrapping this in a try block allows you to catch ValueError and handle it appropriately.

Missing Keys: The code assumes that certain keys will always be present in the JSON response. If the API response format changes or if the data is missing, this can raise a KeyError, which you can catch and handle to avoid crashes.

Unforeseen Errors: Catching a generic Exception ensures that any other unexpected errors are also handled, providing robustness to your function.